### PR TITLE
Added prometheus port to avoid the scraping of port 6443

### PR DESCRIPTION
**Pull request by pierluigilenoci**
**What this PR does / why we need it**:

I need this to prevent Prometheus from scraping port 6443.

You can see an example of the problem from the screenshots attached to this ticket #2860 

**Which issue(s) this PR fixes**:
None directly, address partially the aforementioned issue.

**Special notes for your reviewer**:

None

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @pierluigilenoci. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jklaise**
You can assign the PR to them by writing `/assign @jklaise` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by pierluigilenoci** [comment]
@adriangonz are you able to give me some feedback? Why did the Docs Test fail? I presume I changed nothing related to docs. 🤯

**Comment by axsaucedo** [comment]
@pierluigilenoci just for context, currently docs-build test is actually broken in master, we're investigating the fix

File: helm-charts/seldon-core-operator/templates/deployment_seldon-controller-manager.yaml
Comment (axsaucedo) [inline helm-charts/seldon-core-operator/templates/deployment_seldon-controller-manager.yaml:25]: You don't need this anymore, the reason why we removed it is because prometheus needs to be configured with the rule to scrape from the named port called "metrics". You have to configure prometheus scraping config as per the documentation

**Review by axsaucedo — CHANGES_REQUESTED** [review]
Added a comment, I don't think at this point this is a change that we would be adding given that the current method of scraping is through named ports as opposed to explicit port provisioning via annotation

File: helm-charts/seldon-core-operator/templates/deployment_seldon-controller-manager.yaml
Comment (pierluigilenoci) [reply helm-charts/seldon-core-operator/templates/deployment_seldon-controller-manager.yaml:25]: We use Prometheus' vanilla configuration which comes from the official Helm Chart.
https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/values.yaml#L1565-L1607

With this configuration, Seldon generates an error. See screenshot (https://user-images.githubusercontent.com/36044817/105344911-dd186780-5be3-11eb-955f-ece2d1b029dd.png)

Comment (pierluigilenoci) [reply helm-charts/seldon-core-operator/templates/deployment_seldon-controller-manager.yaml:25]: @axsaucedo can I have a feedback?

**Comment by axsaucedo** [comment]
Thank you for following up @pierluigilenoci - I have discussed with the rest of the team, and the consensus was that we are definitely keen to address the issue of labels and annotations being exposed so that they can be configured through the seldondeployment on the pod level for the serviceOrchestrator and models, but that's still quite different to this PR.

This PR is basically adding the annotation for serviceOrchestrator explicitly for prometheus, but unfortunately this doesn't fully encompass the annotations required for the model pods, so it's more of a workaround as the correct approach here is to add the prometheus configuration. I wanted to provide further context on where we're coming from - does this make sense?

Having said that we'd still be very keen on addressing #2860 and making annotations extensible.

**Comment by pierluigilenoci** [comment]
@axsaucedo it totally makes sense. Do you have an ETA for this?

**Comment by axsaucedo** [comment]
Ok great - will close this one and we can continue discussion on the issue. I will mark the issue as p1, so it can be prioritised in our next sprint planning and we'll have a better idea when this will fit

**Pull request closed without merging**
